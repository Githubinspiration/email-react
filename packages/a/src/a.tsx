import * as React from 'react';

type AElement = React.ElementRef<'a'>;
type RootProps = React.ComponentPropsWithoutRef<'a'>;

export interface AProps extends RootProps {}

export const A = React.forwardRef<AElement, Readonly<AProps>>(
  ({ target = '_blank', style, ...props }, forwardedRef) => (
    <a
      ref={forwardedRef}
      target={target}
      style={{
        color: '#067df7',
        textDecoration: 'none',
        ...style,
      }}
      {...props}
    />
  ),
);

A.displayName = 'A';
