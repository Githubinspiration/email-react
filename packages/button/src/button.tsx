import * as React from 'react';

type ButtonElement = React.ElementRef<'a'>;
type RootProps = React.ComponentPropsWithoutRef<'a'>;

export interface ButtonProps extends RootProps {}

export const Button = React.forwardRef<ButtonElement, Readonly<ButtonProps>>(
  ({ children, style, target = '_blank', ...props }, forwardedRef) => (
    <a
      ref={forwardedRef}
      target={target}
      dangerouslySetInnerHTML={{
        __html: `<!--[if mso]><i style="letter-spacing: 25px;mso-font-width:-100%;mso-text-raise:30pt" hidden>&nbsp;</i><![endif]--><span>${children}</span><!--[if mso]><i style="letter-spacing: 25px;mso-font-width:-100%" hidden>&nbsp;</i><![endif]-->`,
      }}
      style={style}
      {...props}
    />
  ),
);

Button.displayName = 'Button';
