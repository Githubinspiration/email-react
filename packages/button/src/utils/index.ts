export * from "./px-to-pt";
