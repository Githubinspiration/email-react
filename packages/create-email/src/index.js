#!/usr/bin/env node

import { Command } from 'commander';
import { fileURLToPath } from 'node:url';
import fse from 'fs-extra';
import logSymbols from 'log-symbols';
import ora from 'ora';
import path from 'node:path';
import tree from 'tree-node-cli';
import packageJson from '../package.json' assert { type: "json" };

const init = (name) => {
  const spinner = ora('Preparing files...\n').start();

  let projectPath = name;

  if (!projectPath) {
    projectPath = path.join(process.cwd(), 'react-email-starter');
  }
    
  if (typeof projectPath === 'string') {
    projectPath = projectPath.trim()
  }

  const __dirname = path.dirname(fileURLToPath(import.meta.url));
  const templatePath = path.resolve(__dirname, '../template');
  const resolvedProjectPath = path.resolve(projectPath);

  fse.copySync(templatePath, resolvedProjectPath, { recursive: true });

  const fileTree = tree(projectPath, {
    allFiles: true,
    exclude: [/node_modules/],
    maxDepth: 4,
  });

  console.log(fileTree);

  spinner.stopAndPersist({
    symbol: logSymbols.success,
    text: 'React Email Starter files ready',
  });
}

const program = new Command()
  .name(packageJson.name)
  .version(packageJson.version)
  .description(packageJson.description)
  .arguments('[dir]', 'path to initiliaze the project')
  .action(init)
  .parse(process.argv);