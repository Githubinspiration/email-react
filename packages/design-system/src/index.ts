export * from './code';
export * from './icon-button';
export * from './icons';
