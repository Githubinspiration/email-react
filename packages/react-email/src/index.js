#!/usr/bin/env node

const { Command } = require('commander');
const copy = require('copy');
const path = require('path');
const shell = require('shelljs');
const packageJson = require('../package.json');

const program = new Command();

program
  .name(packageJson.name)
  .version(packageJson.version)
  .description(packageJson.description);

const dev = () => {
  const currentPath = `${process.cwd()}/*.tsx`;
  const basePath = path.resolve(__dirname, '../base/emails');
  
  copy(currentPath, basePath, () => {
    shell.cd(path.join(__dirname, '../base'));
    shell.exec('npm run dev', { async: true });
  });
}

program
  .command('dev')
  .action(dev)

program.parse(process.argv);