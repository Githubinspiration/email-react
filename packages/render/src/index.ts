export * from "./render";
export * from "./renderAsync";
