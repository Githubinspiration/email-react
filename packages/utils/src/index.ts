export * from './as';
export * from './copy-text-to-clipboard';
export * from './px-to-pt';
export * from './spaces';
export * from './style-to-string';
export * from './unreachable';
