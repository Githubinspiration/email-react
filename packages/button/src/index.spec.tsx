import * as React from 'react';
import { Button } from './index';
import { render } from '../../render/src/index';

describe('render', () => {
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetModules();
  });

  it('renders the <Button> component', () => {
    const actualOutput = render(<Button href="https://example.com" />);
    expect(actualOutput).toMatchSnapshot();
  });
});
