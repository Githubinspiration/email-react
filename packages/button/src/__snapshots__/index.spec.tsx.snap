// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders the <Button> component 1`] = `"<!DOCTYPE htmlPUBLIC \\"-//W3C//DTD XHTML 1.0 Transitional//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\\"><a target=\\"_blank\\" href=\\"https://example.com\\"><!--[if mso]><i style=\\"letter-spacing: 25px;mso-font-width:-100%;mso-text-raise:30pt\\" hidden>&nbsp;</i><![endif]--><span>undefined</span><!--[if mso]><i style=\\"letter-spacing: 25px;mso-font-width:-100%\\" hidden>&nbsp;</i><![endif]--></a>"`;
